common/crypto/openssl/ec.c

```diff
@@ -320,6 +390,128 @@ oe_result_t oe_ec_generate_key_pair_from_private(
         EVP_PKEY_free(private_pkey);
     return result;
 }
+#else
+oe_result_t oe_ec_generate_key_pair_from_private(
+    oe_ec_type_t curve,
+    const uint8_t* private_key_buf,
+    size_t private_key_buf_size,
+    oe_ec_private_key_t* private_key,
+    oe_ec_public_key_t* public_key)
+{
+    oe_result_t result = OE_UNEXPECTED;
+    EC_GROUP* group = NULL;
+    EVP_PKEY_CTX* public_ctx = NULL;
+    EVP_PKEY_CTX* private_ctx = NULL;
+    EVP_PKEY* public_pkey = NULL;
+    EVP_PKEY* private_pkey = NULL;
+    BIGNUM* private_key_bn = NULL;
+    EC_POINT* point = NULL;
+    uint8_t* buffer = NULL;
+    size_t keysize = 0;
+    OSSL_PARAM_BLD* bld = NULL;
+    OSSL_PARAM* params = NULL;
+
+    if (!private_key_buf || !private_key || !public_key ||
+        private_key_buf_size > OE_INT_MAX)
+        OE_RAISE(OE_INVALID_PARAMETER);
+
+#if !defined(OE_BUILD_ENCLAVE)
+    /* Initialize OpenSSL. */
+    oe_crypto_initialize();
```

Comment:
We don't need this for OpenSSL 3. See https://github.com/openenclave/openenclave/blob/master/host/crypto/openssl/init.c#L13